backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java

```diff
@@ -136,6 +148,29 @@ public TestResultUpload processResultCSV(InputStream csvStream) {
     return csvResult;
   }
 
+  private byte[] translateSpecimenNameToSNOMED(byte[] content, Map<String, String> snomedMap) {
+    String[] rows = new String(content, StandardCharsets.UTF_8).split("\n");
+    String headers = rows[0];
+
+    int specimenTypeIndex =
+        Arrays.stream(headers.split(",")).toList().indexOf(SPECIMEN_TYPE_COLUMN_NAME);
+
+    for (int i = 1; i < rows.length; i++) {
+      var row = rows[i].split(",", -1);
+      var specimenTypeName = Arrays.stream(row).toList().get(specimenTypeIndex).toLowerCase();
+
+      if (!specimenTypeName.matches(ALPHABET_REGEX)) {
+        continue;
+      }
+
+      row[specimenTypeIndex] = snomedMap.get(specimenTypeName);
+
+      rows[i] = String.join(",", row);
+    }
+
+    return String.join("\n", rows).getBytes();
+  }
+
```

Comment:
I'm a little concerned with how many times we iterate through every row and modify the file (validation, processing code and now this).  Out of scope for this PR for sure, but we might want to consider consolidating all the file parsing and modifying.

What I was going to ask is why change it out for the whole file rather than just changing it when we use the value.  But the way you have it will work better for the CSV api so 👍

Reply:
> I'm a little concerned with how many times we iterate through every row and modify the file

Yes 🙃